pkgpanda/actions.py

```diff
@@ -195,9 +195,19 @@ def _start_dcos_target(block_systemd):
 
 def _get_package_list(package_list_id: str, repository_url: str) -> List[str]:
     package_list_url = repository_url + '/package_lists/{}.package_list.json'.format(package_list_id)
-    with tempfile.NamedTemporaryFile() as f:
-        download(f.name, package_list_url, os.getcwd(), rm_on_error=False)
-        package_list = load_json(f.name)
+
+    # TODO(klueska): On Windows there are issues with following a simple
+    # "delete-on-close" semantic, so we instead pass 'delete=False' when we
```

Comment:
Is there (a link to) a description of the problem / symptoms here? It would be nice to be able to clean this up one day, but without knowing how this was observed to deviate from the documented behavior, it will become a "safer to leave messy" problem.

Reply:
The problem is that on Windows you can't call `open()` on a file that's already opened, as it'll raise an exception. Since on Windows we have to close the tempfile after writing before it can be read, `delete=True` renders it useless. Comment updated to clarify.